commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java

```diff
@@ -615,10 +615,12 @@ void register(CommandMetaData<CommandSender> meta, final Argument[] args,
 		}
 
 		List<String> argumentsString = new ArrayList<>();
+		List<AbstractArgument<?, ?, ?, ?>> arguments = new ArrayList<>();
 		for (Argument arg : args) {
+			arguments.add(arg);
 			argumentsString.add(arg.getNodeName() + ":" + arg.getClass().getSimpleName());
 		}
-		RegisteredCommand registeredCommandInformation = new RegisteredCommand(commandName, argumentsString, shortDescription,
+		RegisteredCommand registeredCommandInformation = new RegisteredCommand(commandName, argumentsString, List.copyOf(arguments), shortDescription,
```

Comment:
This looks weird. We have an array of arguments `final Argument[] args`. We iterate over it to add it to a list, then we create an immutable copy of that list. Wouldn't it be better to just go straight to using `List.of(args)`?